src/journal-remote/journal-upload.c

```diff
@@ -156,6 +234,26 @@ static int update_cursor_state(Uploader *u) {
         return log_error_errno(r, "Failed to save state %s: %m", u->state_file);
 }
 
+static int refresh_timeout(Uploader *u) {
+        int r;
+        u->bytes_left = arg_batch_max_bytes;
+        u->entries_left = arg_batch_max_entries;
+        u->uploading = true;
+
+        if (arg_batch_timeout_usec != USEC_INFINITY) {
+                struct itimerspec its;
+                its.it_value.tv_sec = arg_batch_timeout_usec / USEC_PER_SEC;
+                its.it_value.tv_nsec = (arg_batch_timeout_usec % USEC_PER_SEC) * 1000;
+                its.it_interval.tv_sec = 0;
+                its.it_interval.tv_nsec = 0;
```

Comment:
please use structured initialization, i.e.: 

```c
struct itimerspec its = {};
```

And then use timespec_load() to convert from usec_t to timespec